src/style/style.js

```diff
@@ -385,6 +391,21 @@ class Style extends Evented {
         return false;
     }
 
+    get order(): Array<string> {
+        if (this.map._optimizeForTerrain && this.terrain) {
+            if (!this._drapedFirstOrder.length) {
+                this._updateDrapeFirstLayers();
```

Comment:
It seems like this should never get hit bcoz `_updateDrapeFirstLayers` should be called whenever layer order within `Style` is mutated, should this just be an assert for `this._drapedFirstOrder.length` instead?

Reply:
Good call. Updated for assert in https://github.com/mapbox/mapbox-gl-js/pull/10258/commits/b732630d4465ce232a58c71932843833dc2885e9.